### DISCUSSION TITLE
I don't want my bot to reply to offline messages

### DISCUSSION THREAD

**Discussion by MrsSAS** [eternnoir/pyTelegramBotAPI#1676] [Q&A] [answered]
When restarting my bot, it replies to all received messages since it was offline.
Reading the parameter description I saw that:
`bot.polling(interval=1, skip_pending=True)
`
only replies to the last message. Investigating further, the `skip_pending` parameter is linked to the function `get_updates(offset=-1)`, which makes sense, in fact when I set `offset=-2` the bot replies to the last 2 messages, and so on...
But, what if I wanted to prevent the bot from repling any old messages? Because when I set `offset=0`, it still responds to all of them.
How can I solve this?

**Comment by coder2020official** [comment, marked as answer]
Hey!
Using skip_pending processes only last update, yes.
If you are so excited about skipping ALL updates, you can use [delete_webhook](https://pytba.readthedocs.io/en/latest/sync_version/index.html#telebot.TeleBot.delete_webhook), by passing drop_pending_updates=True.

**Suggestion to your code**
I recommend using bot.infinity_polling() instead of just polling. If you use simple polling, your bot may stop on system exception.
With infinity polling, this won't take place.

**Reply by MrsSAS** [reply]
Thanks for the suggestion but (in anticipation to implement a webhook), don't you think `bot.infinity_polling()` would spam a little bit too much to telegram servers?

**Reply by coder2020official** [reply]
yeah, it will spam telegram servers. But telegram servers allow that for now. You can use webhooks instead if you don't want to use polling

**Reply by MrsSAS** [reply]
Great, thanks a lot! Just one last question, among the different alternatives I can read on [this page](https://github.com/eternnoir/pyTelegramBotAPI/tree/master/examples/webhook_examples) which one would you suggest to me for implementing a webhook (from your own experience)?

**Reply by coder2020official** [reply]
You can use run_webhooks (built-in func) which starts app using fastapi.
For smth more complicated i recommend using fastapi itself or any other framework such as flask
